### PR TITLE
Support Rails 5.0

### PR DESCRIPTION
**Pull request by yahonda**
This pull request supports Rails 5.0 and Oracle enhanced adapter 1.7 supporting Rails 5.

I have run `rake spec` locally then most of specs are fine except for #114 which also reproduces with Rails 4.2.

### CODE REVIEW

**Comment by javornikolov** [comment]
> I have run rake spec locally then most of specs are fine except for #114 which also reproduces with Rails 4.2.

@yahonda, is that trouble also with Oracle 11.2 or with Oracle 12 only?

File: Gemfile [hunk 1]
```diff
@@ -10,8 +10,8 @@ group :test, :development do
   gem 'rspec', '~> 3.1'
 
   unless ENV['NO_ACTIVERECORD']
-    gem 'activerecord', '>= 3.2.3', '< 4.3.0'
-    gem 'activerecord-oracle_enhanced-adapter', '>= 1.4.1', '< 1.7.0'
+    gem 'activerecord', '>= 3.2.3', '< 5.1.0'
```
Comment (javornikolov) [inline Gemfile]: Looks like active record 5 doesn't support older versions of Ruby (it needs >= 2.2.2). We should somehow limit Rails to v4 for the old versions of Ruby. (At least for the purpose of Travis CI builds).

**Comment by yahonda** [comment]
Thanks for the update. #114 reproduces only with Oracle Database 12c (12.1.0.2), It does not reproduce with Oracle Database 11g (11.2.0.2.0).

File: Gemfile (hunk 1 above)
Comment (yahonda) [inline Gemfile]: Yes. [Ruby 2.2.2+ required](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#ruby-2-2-2-required) to run Rails 5.0. I am not familiar with limiting ruby version per Rails version, though.

Comment (yahonda) [inline Gemfile]: travis.yml has a feature called [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#Build-Matrix) and I've found sprockets-rails repository is using this feature https://github.com/rails/sprockets-rails/tree/master/gemfiles

**Comment by javornikolov** [comment]
> travis.yml has a feature called Build Matrix and I've found sprockets-rails repository is using this feature https://github.com/rails/sprockets-rails/tree/master/gemfiles

Thanks @yahonda! I just merged into master a basic matrix configuration. Could you please update it here accordingly for the Rails 5 support (and let's keep a gemfile for the 4.2 setup too).

**Comment by yahonda** [comment]
Updated this pull request.

**Comment by javornikolov** [comment]
Nice :-) Thank you @yahonda!
We end up with multiple similar gemfiles to maintain. I'm not sure if there is a clean way to reduce that duplication.

**Comment by yahonda** [comment]
Thanks for merging. Yes, there are some duplicates including the "ruby-plsql.gemspec" file. So far I do not have enough time to tackle them...

**Pull request merged**